exercises/ex4.Rnw

```diff
@@ -93,72 +93,85 @@ mark_my_assignment()
+Choose the amount of digits you report for the mean and
+quantiles based on the MCSEs such that you leave out digits where
+MCSE is nonzero (Example: if posterior mean is 2.345678 and MCSE is 0.0012345, report two digits after the decimal sign, taking into account the usual rounding rule, so you would report 2.35. Further digits do not contain useful information due to the Monte Carlo uncertainty.).
+\begin{itemize}
+\item \textbf{Note!} The answer is graded as correct only if the number of digits
+reported is correct! The number of significant digits can be different for the mean
+and quantile estimates.
+\item \textbf{Hint!} Quantiles can be computed with the \texttt{quantile} function. With $S$ draws, the MCSE for $\text{E}[\theta]$ is $\sqrt{\text{Var} [\theta]/S}$.
+MCSE for the quantile estimates can be computed with the
+\texttt{mcse\_quantile} function from the {\tt aaltobda} package.
+\end{itemize}
 
-\item[c)] Implement a function in R for computing the \textbf{logarithm} of the density of the posterior for arbitrary values of $\alpha$ and $\beta$ and data $x$, $y$ and $n$. Below is an example of how the function should be named and work if you want to check them with \texttt{markmyassignment}. 
- 
-<<echo=FALSE>>=
-p_log_posterior <- function(alpha, beta, x, y, n) -15.78798
-@
-  
-<<echo=TRUE>>=
-library(aaltobda)
-data("bioassay")
-p_log_posterior(alpha = 3, 
-                beta = 9, 
-                x = bioassay$x, 
-                y = bioassay$y, 
-                n = bioassay$n)
-@  
-
-\textbf{Hint!} Equation (3.16) in the course book. The \textbf{logarithm} of the prior density was already implemented in b). For computing the \textbf{logarithm} of the likelihood, use the {\tt bioassaylp} function from the {\tt aaltobda} package. The data can be loaded with the R command {\tt data("bioassay")}.\\
-\textbf{Hint!} Logarithm of the product of two densities is the sum of the log-densities, i.e. $\log ab = \log a + \log b$.
-
-\item[d)] Plot the posterior density in a grid of points ($\alpha \in [-4,4]$, $\beta \in [-10,30]$) using the {\tt bioassay\_posterior\_density\_plot} function from the {\tt aaltobda} package. Pass the {\tt p\_log\_posterior} function you implemented in c) as argument to {\tt bioassay\_posterior\_density\_plot}.
-\item[e)] Implement two functions in: 1) A function for computing the log importance ratios (log importance weights) for draws from the prior distribution in 1a) when the target distribution is the posterior distribution. 2) A function for exponentiating the log importance ratios and normalizing them to sum to one. Below is a test example, the functions can also be tested with \texttt{markmyassignment}. \textbf{Note!} The values below are \emph{only} a test case, you need to use correct draws for {\tt alpha} and {\tt beta} in the final report. \\
+\end{itemize}
+\item[] {\bf Importance sampling}. Now we discard our posterior draws and
```

Comment:
is `[]` a typo?

Reply:
Not really, it just removed the numbering from enumerate. Without numbering the enumerate is unnecessary, so I removed it